test/zIntegrationTests.ts

```diff
@@ -127,3 +162,76 @@ describe("Test suit for setting up the stage", function () {
         expect(await r.leverageAllocator.getAddressToLvUSDAvailable(await user.getAddress())).to.equal(parseUnitsNum(initialUserLevAllocation));
     });
 });
+
+describe("Test suit for getting leverage", function () {
+    let leverageUserIsTaking: number;
+
+    before(async function () {
+        await setupEnvForIntegrationTests();
+        leverageUserIsTaking = getFloatFromBigNum(
+            await r.parameterStore.getAllowedLeverageForPosition(userOUSDPrincipleInEighteenDecimal, numberOfCycles));
+        console.log("Will take %s lev while user has allocation of %s", leverageUserIsTaking,
+            getFloatFromBigNum(await r.leverageAllocator.getAddressToLvUSDAvailable(user.address)));
+
+        await approveAndGetLeverageAsUser(userOUSDPrincipleInEighteenDecimal, numberOfCycles, r, user);
+        positionId = 0;
+    });
+
+    it("Should have created a single position and assign it to user", async function () {
+        printPositionState(r, positionId);
+        printPoolState(r.curveLvUSDPool);
+        printMiscInfo(r, user);
+        const nftBalance = await r.positionToken.balanceOf(user.address);
+        expect(nftBalance).to.equal(1);
+    });
+
+    it("Should have assigned origination fee to treasury", async function () {
+        const treasuryBalance = getFloatFromBigNum(
+            await r.externalOUSD.balanceOf(r.treasurySigner.address),
+        );
+        /// origination fee is 5% at the moment, lvUSDBorrowed is 171 so 5% of it is roughly 8.5
+        expect(treasuryBalance).to.closeTo(8.5, 0.1);
+    });
+
+    it("Should have deposited leverage and principle in vault (minus fee)", async function () {
+        const vaultOUSDBalance = getFloatFromBigNum(await r.vault.totalAssets());
+        /// this should be equal to principle + leveraged OUSD - fees = 100 + 171 - 8.5 = 262.5
+        expect(vaultOUSDBalance).to.closeTo(262, 1);
+    });
+
+    it("Should have reduced lev allocation", async function () {
+        const allowedLev = getFloatFromBigNum(await r.leverageAllocator.getAddressToLvUSDAvailable(user.address));
+        expect(allowedLev).to.equal(initialUserLevAllocation - leverageUserIsTaking);
+    });
+});
+
+describe("test suit for rebase events", function () {
+    const rebaseAmount = 20;
+    before(async function () {
+        await setupEnvForIntegrationTests();
+        await approveAndGetLeverageAsUser(userOUSDPrincipleInEighteenDecimal, numberOfCycles, r, user);
+        positionId = 0;
+        // at this stage we have a position created. Now simulating a rebase
+        await r.externalOUSD.connect(pretendOUSDRebaseSigner).transfer(r.vault.address, parseUnitsNum(rebaseAmount));
```

Comment:
where is `pretendOUSDRebaseSigner` defined?

Reply:
https://github.com/thisisarchimedes/Archimedes_Finance/blob/60c3b60033e9a18c5ab31a08dc1d3a45b68b9935/test/zIntegrationTests.ts#L80